### PR TITLE
Update speed library

### LINKED ISSUE #1176: codahale/hdrhistogram repo url has been transferred under the github HdrHstogram umbrella
## Problem 

The codahale/hdrhistogram repo has been transferred under the github HdrHstogram umbrella with the help from the original author in Sept 2020 (new repo url https://github.com/HdrHistogram/hdrhistogram-go). The main reasons are to group all implementations under the same roof and to provide more active contribution from the community as the original repository was archived several years ago.

The dependency URL should be modified to point to the new repository URL. The tag "v0.9.0" was applied at the point of transfer and will reflect the exact code that was frozen in the original repository.

If you are using Go modules, you can update to the exact point of transfer using the @v0.9.0 tag in your go get command.
```
go mod edit -replace github.com/codahale/hdrhistogram=github.com/HdrHistogram/hdrhistogram-go@v0.9.0
```

## Performance Improvements

From the point of transfer, up until now (mon 16 aug 2021), we've released 3 versions that aim support the standard HdrHistogram serialization/exposition formats, and deeply improve READ performance. 
We recommend to update to the latest version.

### PR DESCRIPTION
**Pull request by sagikazarmark**
Related #1127 

Fixes #1176

### CODE REVIEW

File: go.sum [hunk 1]
```diff
@@ -521,4 +554,5 @@ gopkg.in/yaml.v3 v3.0.0-20210107192922-496545a6307b h1:h8qDotaEPuJATrMmW04NCwg7v
 gopkg.in/yaml.v3 v3.0.0-20210107192922-496545a6307b/go.mod h1:K4uyk7z7BCEPqu6E+C64Yfv1cQ7kz7rIZviUmN+EgEM=
 honnef.co/go/tools v0.0.0-20190102054323-c2f93a96b099/go.mod h1:rf3lG4BRIbNafJWhAfAdb/ePZxsR/4RtNHQocxwk9r4=
 honnef.co/go/tools v0.0.0-20190523083050-ea95bdfd59fc/go.mod h1:rf3lG4BRIbNafJWhAfAdb/ePZxsR/4RtNHQocxwk9r4=
+rsc.io/pdf v0.1.1/go.mod h1:n8OzWcQ6Sp37PL01nO98y4iUCRdTGarVfzxY20ICaU4=
```
Comment (peterbourgon) [inline go.sum:557]: Why is this here?

File: go.sum [hunk 2]
```diff
@@ -9,6 +13,7 @@ github.com/VividCortex/gohistogram v1.0.0 h1:6+hBz+qvs0JOrrNhhmR7lFxo5sINxBCGXrd
 github.com/VividCortex/gohistogram v1.0.0/go.mod h1:Pf5mBqqDxYaXu3hDrrU+w6nw50o/4+TcAqDqk/vUH7g=
 github.com/afex/hystrix-go v0.0.0-20180502004556-fa1af6a1f4f5 h1:rFw4nCn9iMW+Vajsk51NtYIcwSTkXr+JGrMd36kTDJw=
 github.com/afex/hystrix-go v0.0.0-20180502004556-fa1af6a1f4f5/go.mod h1:SkGFH1ia65gfNATL8TAiHDNxPzPdmEL5uirI2Uyuz6c=
+github.com/ajstarks/svgo v0.0.0-20180226025133-644b8db467af/go.mod h1:K08gAheRH3/J6wwsYMMT4xOr94bZjxIelGM0+d/wbFw=
```
Comment (peterbourgon) [inline go.sum:16]: Why is this here?

File: go.sum (hunk 1 above)
Comment (sagikazarmark) [reply go.sum:557]:
```
❯ go mod graph | grep rsc.io/pdf
gonum.org/v1/plot@v0.0.0-20190515093506-e2840ee46a6b rsc.io/pdf@v0.1.1
❯ go mod graph | grep gonum.org/v1/plot@v0.0.0-20190515093506-e2840ee46a6b
gonum.org/v1/gonum@v0.8.2 gonum.org/v1/plot@v0.0.0-20190515093506-e2840ee46a6b
gonum.org/v1/plot@v0.0.0-20190515093506-e2840ee46a6b github.com/ajstarks/svgo@v0.0.0-20180226025133-644b8db467af
gonum.org/v1/plot@v0.0.0-20190515093506-e2840ee46a6b github.com/fogleman/gg@v1.2.1-0.20190220221249-0403632d5b90
gonum.org/v1/plot@v0.0.0-20190515093506-e2840ee46a6b github.com/golang/freetype@v0.0.0-20170609003504-e2365dfdc4a0
gonum.org/v1/plot@v0.0.0-20190515093506-e2840ee46a6b github.com/jung-kurt/gofpdf@v1.0.3-0.20190309125859-24315acbbda5
gonum.org/v1/plot@v0.0.0-20190515093506-e2840ee46a6b golang.org/x/exp@v0.0.0-20180807140117-3d87b88a115f
gonum.org/v1/plot@v0.0.0-20190515093506-e2840ee46a6b golang.org/x/image@v0.0.0-20180708004352-c73c2afc3b81
gonum.org/v1/plot@v0.0.0-20190515093506-e2840ee46a6b gonum.org/v1/gonum@v0.0.0-20180816165407-929014505bf4
gonum.org/v1/plot@v0.0.0-20190515093506-e2840ee46a6b rsc.io/pdf@v0.1.1
❯ go mod graph | grep gonum.org/v1/gonum@v0.8.2
github.com/HdrHistogram/hdrhistogram-go@v1.1.0 gonum.org/v1/gonum@v0.8.2
gonum.org/v1/gonum@v0.8.2 golang.org/x/exp@v0.0.0-20190125153040-c74c464bbbf2
gonum.org/v1/gonum@v0.8.2 golang.org/x/tools@v0.0.0-20190206041539-40960b6deb8e
gonum.org/v1/gonum@v0.8.2 gonum.org/v1/netlib@v0.0.0-20190313105609-8cb42192e0e0
gonum.org/v1/gonum@v0.8.2 gonum.org/v1/plot@v0.0.0-20190515093506-e2840ee46a6b
```

The new histogram library requires it for some reason, but since it's only the mod file I doubt it ever gets used by any code path that Go kit uses.

File: go.sum (hunk 2 above)
Comment (sagikazarmark) [reply go.sum:16]:
```
❯ go mod graph | grep github.com/ajstarks/svgo
gonum.org/v1/plot@v0.0.0-20190515093506-e2840ee46a6b github.com/ajstarks/svgo@v0.0.0-20180226025133-644b8db467af
```

Same answer as above.

File: go.sum (hunk 1 above)
Comment (sagikazarmark) [reply go.sum:557]: Quick search shows it's only imported in a test: https://github.com/HdrHistogram/hdrhistogram-go/search?q=gonum

God, I hate Go for this.

**Comment by sagikazarmark** [comment]
@peterbourgon I don't think the added entries in go.sum is a problem. Since they are just checksums for the the mod files, not the libraries themselves, Go only downloads the mod files for these transitive dependencies, but they are never compiled into the end result.

I agree that this is confusing, irritating to see even, but that's how Go modules work today. :\

**Review by peterbourgon — APPROVED** [review]
